src/index/ivf/ivf.cc

```diff
@@ -197,12 +197,8 @@ IvfIndexNode<T>::Build(const DataSet& dataset, const Config& cfg) {
     const BaseConfig& base_cfg = static_cast<const IvfConfig&>(cfg);
     auto build_thread_num = base_cfg.get_build_thread_num();
     ThreadPool::ScopedOmpSetter setter(build_thread_num);
-    auto err = Train(dataset, cfg);
-    if (err != Status::success) {
-        return err;
-    }
-    err = Add(dataset, cfg);
-    return err;
+    RETURN_IF_ERROR(Train(dataset, cfg));
```

Comment:
suggest to change like this:
    RETURN_IF_ERROR(Train(dataset, cfg));
    RETURN_IF_ERROR(Add(dataset, cfg));
    return Status::success;

Reply:
Chatted offline, agreed to keep it as is